src/compas/scene/sceneobject.py

```diff
@@ -26,6 +27,10 @@ class SceneObject(object):
     # add this to support the descriptor protocol vor Python versions below 3.6
     __metaclass__ = DescriptorProtocol
 
+    def __new__(cls, item, **kwargs):
+        cls = get_sceneobject_cls(item, **kwargs)
```

Comment:
maybe add an underscore or use a different name so it is clear on the next line that you are not overwriting `cls` by accident